unity-renderer/Assets/Scripts/MainScripts/DCL/EmoteAnimationsPlugin/EmoteAnimationLoader.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Threading;
+using AvatarSystem;
+using Cysharp.Threading.Tasks;
+using UnityEngine;
+
+namespace DCL.Emotes
+{
+    public class EmoteAnimationLoader : IEmoteAnimationLoader
+    {
+        private readonly IWearableRetriever retriever;
+        public AnimationClip animation { get; internal set; }
+
+        public EmoteAnimationLoader(IWearableRetriever retriever) { this.retriever = retriever; }
+
+        public async UniTask LoadEmote(GameObject container, WearableItem emote, string bodyShapeId, CancellationToken ct = default)
+        {
+            if (container == null)
+                throw new NullReferenceException("Container cannot be null");
+
+            if (emote == null)
+                throw new NullReferenceException("Emote cannot be null");
+
+            if (string.IsNullOrEmpty(bodyShapeId))
+                throw new NullReferenceException("bodyShapeId cannot be null or empty");
+
+            ct.ThrowIfCancellationRequested();
+
+            WearableItem.Representation representation = emote.GetRepresentation(bodyShapeId);
+            if (representation == null)
+            {
+                throw new Exception($"No representation for {bodyShapeId} of emote: {emote.id}");
+            }
+
+            await retriever.Retrieve(container, emote.GetContentProvider(bodyShapeId), emote.baseUrlBundles, representation.mainFile, ct);
+
+            animation = retriever.rendereable.container.GetComponentInChildren<Animation>()?.clip;
```

Comment:
you could use the renderable returned by `retriever.Retrieve` instead of accessing the property

Reply:
I prefered this so I can just mock the property (which is easier) in tests, instead of overriding the method using `Args.Any<>` which I dont like that much.

But I have no strong opinions against this change.

Reply:
i think it feels unnatural to use the result of a function you just called as a separated property

Reply:
Fair enough, pushed the change